util/topgen/c.py

```diff
@@ -48,17 +48,24 @@ def as_c_type(self):
 
 
 class MemoryRegion(object):
-    def __init__(self, name, base_addr, size_bytes):
+    def __init__(self, name, addr, size_bytes):
         self.name = name
-        self.base_addr = base_addr
+        self.addr = addr
```

Comment:
Why this rename? Isn't this the address of the start of the memory region? That's usually called a base address.

Reply:
Oh, its because now from this field it can generate base address defines, and also offset defines. I have added another helper to this routine to generate offset defines.

Reply:
But this field is the address (relative or absolute) of the base of the memory region. I don't think your other changes affect this, so I think the rename is probably wrong.

Reply:
Happy to revert this part of the change.

Reply:
Done.